### PR TITLE
br: replace exp/slices with std

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #45933

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: br/pkg/lightning/lightning.go [hunk 1]
```diff
@@ -960,8 +961,8 @@ func checkSystemRequirement(cfg *config.Config, dbsMeta []*mydump.MDDatabaseMeta
 				tableTotalSizes = append(tableTotalSizes, tb.TotalSize)
 			}
 		}
-		slices.SortFunc(tableTotalSizes, func(i, j int64) bool {
-			return i > j
+		slices.SortFunc(tableTotalSizes, func(i, j int64) int {
+			return -cmp.Compare(i, j)
```
Comment (D3Hunter) [inline br/pkg/lightning/lightning.go:965]:
```suggestion
			return cmp.Compare(j, i)
```

File: br/pkg/restore/db.go [hunk 2]
```diff
@@ -403,8 +404,8 @@ func (db *DB) ensureTablePlacementPolicies(ctx context.Context, tableInfo *model
 // FilterDDLJobs filters ddl jobs.
 func FilterDDLJobs(allDDLJobs []*model.Job, tables []*metautil.Table) (ddlJobs []*model.Job) {
 	// Sort the ddl jobs by schema version in descending order.
-	slices.SortFunc(allDDLJobs, func(i, j *model.Job) bool {
-		return i.BinlogInfo.SchemaVersion > j.BinlogInfo.SchemaVersion
+	slices.SortFunc(allDDLJobs, func(i, j *model.Job) int {
+		return -cmp.Compare(i.BinlogInfo.SchemaVersion, j.BinlogInfo.SchemaVersion)
```
Comment (D3Hunter) [inline br/pkg/restore/db.go:408]: doiitto

**Review by D3Hunter — COMMENTED** [review]
rest lgtm

File: br/pkg/lightning/lightning.go (hunk 1 above)
Comment (hawkingrei) [reply br/pkg/lightning/lightning.go:965]: I think what he wants here is to sort in descending order.

Comment (lance6716) [reply br/pkg/lightning/lightning.go:965]: the arguments are also swapped. It's about readability.

File: br/pkg/restore/db.go (hunk 2 above)
Comment (hawkingrei) [reply br/pkg/restore/db.go]: Done

**Review by D3Hunter — APPROVED** [review]

**Pull request merged**